HeterogeneousCore/CUDAServices/test/testCUDAService.cpp

```diff
@@ -29,13 +30,10 @@ TEST_CASE("Tests of CUDAService", "[CUDAService]") {
 
   // Test setup: check if a simple CUDA runtime API call fails:
   // if so, skip the test with the CUDAService enabled
-  int deviceCount = 0;
-  auto ret = cudaGetDeviceCount( &deviceCount );
+  int deviceCount = supportedCudaDevices().size();
 
-  if( ret != cudaSuccess ) {
-    WARN("Unable to query the CUDA capable devices from the CUDA runtime API: ("
-         << ret << ") " << cudaGetErrorString( ret ) 
-         << ". Running only tests not requiring devices.");
+  if (deviceCount == 0) {
+    WARN("No supported CUDA devices available. Running only tests not requiring devices.");
```

Comment:
Could this be changed to use `existSansCUDADevices()`? Or, should that be changed to use `supportedCudaDevices()` as well?

Reply:
Good point.

Yes, if we do not build for any of the available devices, we should skip those tests as well.

Reply:
What about using the `exitSansCUDADevices()` here? Or do you prefer to return to that later?

Reply:
I am not very familiar with how the test harness works, but if I understand the intent, you wanted to run part of the tests even if there are no available devices:
```c++
    SECTION("Enabled only if there are CUDA capable GPUs") {
      auto cs = makeCUDAService(ps, ar);
      if(deviceCount <= 0) {
        REQUIRE(cs.enabled() == false);
        WARN("CUDAService is disabled as there are no CUDA GPU devices");
      }
      else {
        REQUIRE(cs.enabled() == true);
        INFO("CUDAService is enabled");
      }
    }
```

If we call `exitSansCUDADevices()` the test would never reach that part.

Reply:
Right, so I was not able read my own code.